### PR TITLE
feat: add version and version picker

### PR DESCRIPTION
**Pull request by tinayuangao**
- add version picker (using menu)
- make stackblitz use the same version as the doc site

Demo: https://material2-docs-dev.firebaseapp.com/

### CODE REVIEW

File: src/app/shared/version/version.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/** Doc site versions. We update the urls and titles manually */
+// TODO(tinayuangao): Update the url with material.angular.io domain
+export const docVersions: VersionInfo[] = [
+  {
+    url: 'https://material2-docs-5.firebaseapp.com/',
+    title: '5.2.4'
+  },
+  {
+    url: `http://material2-docs-dev.firebaseapp.com`,
```
Comment (mmalerba) [inline src/app/shared/version/version.ts]: is this supposed to be pointed at the dev one?

File: src/app/shared/version-picker/version-picker.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@Component({
+  selector: 'version-picker',
+  templateUrl: './version-picker.html'
+})
+export class VersionPicker {
+  docVersions = docVersions;
+
+  materialVersion = materialVersion;
+
+  displayVersion = materialVersion.match(/\d+.\d+.\d+/)[0];
```
Comment (mmalerba) [inline src/app/shared/version-picker/version-picker.ts]: extract the regex into a var

File: src/app/shared/version/version.ts (hunk 1 above)
Comment (tinayuangao) [reply src/app/shared/version/version.ts]: No... I will update the url once we set up the domain for v5.

File: src/app/shared/version-picker/version-picker.ts (hunk 2 above)
Comment (tinayuangao) [reply src/app/shared/version-picker/version-picker.ts]: Done. Thanks for review!

**Comment by tinayuangao** [comment]
Updated the pr with actual urls. 
Added npm scripts for v5-dev and v5-prod.

The v5 is available https://v5.material.angular.io/ (without version picker)

File: package.json [hunk 3]
```diff
@@ -15,7 +15,9 @@
     "prod-build": "npm run build-themes && ng build --aot --prod",
     "postinstall": "webdriver-manager update && bash ./tools/fetch-assets.sh",
     "publish-prod": "npm run build-themes && ng build --aot --prod && firebase use material-angular-io && firebase deploy",
-    "publish-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-dev && firebase deploy"
+    "publish-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-dev && firebase deploy",
+    "publish-v5-prod": "npm run build-themes && ng build --aot --prod && firebase use v5-material-angular-io && firebase deploy",
+    "publish-v5-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-5 && firebase deploy"
```
Comment (jelbourn) [inline package.json]: File an issue saying we should create a deploy script for this rather than adding more npm commands over time?

Comment (tinayuangao) [reply package.json]: Added file `tools/deploy.sh` and updated `package.json`

**Pull request merged**